packages/components/menu/src/menu-group.tsx

```diff
@@ -1,24 +1,68 @@
 import { HTMLChakraProps, chakra, forwardRef } from "@chakra-ui/system"
 import { cx } from "@chakra-ui/shared-utils"
+import { useVirtualizer } from "@tanstack/react-virtual"
 
 import { useMenuStyles } from "./menu"
+import { useRef } from "react"
 
-export interface MenuGroupProps extends HTMLChakraProps<"div"> {}
+export interface MenuGroupProps extends HTMLChakraProps<"div"> {
+  virtualized?: boolean
+  virtualizedOptions?: {
+    rowHeight: number
+    containerHeight: number
+  }
+}
 
 export const MenuGroup = forwardRef<MenuGroupProps, "div">((props, ref) => {
   const { title, children, className, ...rest } = props
+  const arrayChildren = children as any[] // FIX this
 
+  const virtualizedRef = useRef(null)
   const _className = cx("chakra-menu__group__title", className)
   const styles = useMenuStyles()
 
+  const rowVirtualizer = useVirtualizer({
+    count: arrayChildren?.length,
+    getScrollElement: () => virtualizedRef.current,
+    estimateSize: () => props.virtualizedOptions?.rowHeight ?? 35,
+  })
+
   return (
     <div ref={ref} className="chakra-menu__group" role="group">
       {title && (
         <chakra.p className={_className} {...rest} __css={styles.groupTitle}>
           {title}
         </chakra.p>
       )}
-      {children}
+      {props.virtualized ? (
+        <chakra.div
+          ref={virtualizedRef}
+          height={props.virtualizedOptions?.containerHeight ?? 400}
```

Comment:
it does not make sense to have a `virtualizedOptions` without the `virtualized` props, maybe we should consider add a bit of logic to avoid those `??`